### DISCUSSION TITLE
What is the best way to learn OpenSceneGraph?

### DISCUSSION THREAD

**Discussion by kahlenberg** [openscenegraph/OpenSceneGraph#1270] [General]
I could not find any useful online resource. I want to use osgEarth which is based on OSG but there is no documentation.

**Comment by robertosfield** [comment]
No documentation except for over a hundreds examples, several books, doxygen reference docs. and all the history of discussions on the public forums.

However, while there is resources to learn the OSG I would recommend just skipping OSG/OpenGL that are no just in maintenance mode for legacy applications, and just move to the VulkanSceneGraph/Vulkan both of while are modern APIs for modern hardware and in current development.  There are alternatives to osgEarth in development like vsgCs and Rocky.

**Comment by xarray** [comment]
https://www.packtpub.com/product/openscenegraph-30-beginners-guide/9781849512824
https://www.packtpub.com/product/openscenegraph-3-cookbook/9781849516884

hopefully these books are still available :)
